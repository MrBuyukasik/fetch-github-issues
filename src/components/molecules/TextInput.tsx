import React from 'react';
import {View} from 'react-native';

import styled from 'styled-components/native';
import Input from '../atoms/Input';
import Text from '../atoms/Text';

const StyledView = styled(View)`
  margin: 16px;
`;

const StyledText = styled(Text)`
  margin: 6px;
`;

interface Props {
  label: string;
  defaultValue?: string;
  onUpdate(value: string): void;
  testID?: string;
}

const TextInput = (props: Props) => {
  const {label, defaultValue, onUpdate, testID} = props;

  return (
    <StyledView>
      <StyledText testID={testID} label={label} />
      <Input defaultValue={defaultValue} onUpdate={onUpdate} />
    </StyledView>
  );
};

export default TextInput;
