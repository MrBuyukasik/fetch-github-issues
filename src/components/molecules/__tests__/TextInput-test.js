import React from 'react';
import {render} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import TextInput from '../TextInput';

describe('Header test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<TextInput label="Facebook" style={{}} onUpdate={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders Labels', () => {
    const {getByText} = render(
      <TextInput
        label="Enter Organization name"
        style={{}}
        onUpdate={jest.fn()}
      />,
    );
    expect(getByText('Enter Organization name')).toHaveTextContent(
      'Enter Organization name',
    );
  });
});
