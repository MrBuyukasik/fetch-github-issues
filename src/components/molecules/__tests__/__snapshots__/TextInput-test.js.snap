// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 16,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "width": "100%",
        },
        Object {
          "marginBottom": 6,
          "marginLeft": 6,
          "marginRight": 6,
          "marginTop": 6,
        },
      ]
    }
  >
    Facebook
  </Text>
  <TextInput
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#000000",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 1,
          "fontSize": 18,
          "height": 50,
          "marginBottom": 2,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 2,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
